### PR TITLE
Add Action for publishing Playwright reports

### PR DESCRIPTION
**Pull request by sunker**
This PR adds a new Action for publishing Playwright reports to GCS.

### CODE REVIEW

File: publish-report/action.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+      uses: 'google-github-actions/setup-gcloud@v2'
+
+    - name: Publish report to Cloud Storage
+      shell: bash
+      run: |
+        echo "Publish release to Google Cloud Storage:"
+        gsutil cp -r ${{ inputs.directory }}. gs://releng-pipeline-artifacts-dev/${{ github.event.repository.name }}/${{ github.event.number }}/Grafana-v${{ matrix.GRAFANA_VERSION }}
+
+    - name: Write report link to summary
+      shell: bash
```
Comment (tolzhabayev) [inline publish-report/action.yml:34]: just curious, what does `shell: bash` do in this context / why is it necessary?

Comment (sunker) [reply publish-report/action.yml:34]: To run commands in [composite actions](https://docs.github.com/en/actions/creating-actions/creating-a-composite-action#creating-an-action-metadata-file) you need to specify `shell`. Also see https://github.com/orgs/community/discussions/18597.

**Review by academo — CHANGES_REQUESTED** [review]
@sunker since we are putting the effort to make this action available as opensource source code there is the possibility to modify this current one to maybe allow someone outside our organization to reuse it:

* allow to pass the bucket we'll use as a parameter
* allow to pass the credential to authenticate as a parameter too, then where we use it in our repositories we'll pass the secret we have stored.
* Generate the links and paths based on those parameters.

**Comment by sunker** [comment]
@academo I don't think we want to do that. Everything in this action is very much Grafana plugin specific.

>allow to pass the bucket we'll use as a parameter

The vault secret for the `grafana/artifacts-uploader-service-account` service account only has permissions to access this particular bucket. This is good, as Playwright reports may be heavy so we want to control bucket retention period, avoid plugin authors overwriting other plugin artifacts etc. 

>allow to pass the credential to authenticate as a parameter too, then where we use it in our repositories we'll pass the secret we have stored.

There are already many Actions out there that can push files to GCS. If a certain grafana squad wants to use their own bucket, the can just use one of those. The purpose of this Actions is to capture the generic use case. 

Docs are clear that this Action only works for repos that are in the Grafana org.

File: publish-report/README.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
+# Publish Playwright report
```
Comment (leventebalogh) [inline publish-report/README.md:1]: As this is kind of a "private" action, would it make sense to put an admonition to the top of the README to "underline" this? E.g.:
```md
> [!IMPORTANT]  
> This action is only for repositories under the `grafana` organisation.
```

Comment (sunker) [reply publish-report/README.md:1]: Yes I think that makes sense, will update the readme. thanks @leventebalogh

**Review by academo — APPROVED** [review]

**Pull request merged**